editor/sidebar/post-visibility/index.js

```diff
@@ -106,38 +107,39 @@ class PostVisibility extends Component {
 		return (
 			<div className="editor-post-visibility">
 				<span>{ __( 'Visibility' ) }</span>
-				<button className="editor-post-visibility__toggle button-link" onClick={ this.toggleDialog }>
-					{ getVisibilityLabel( visibility ) }
-				</button>
-
-				{ this.state.opened &&
-					<div className="editor-post-visibility__dialog">
-						<div className="editor-post-visibility__dialog-arrow" />
-						<div className="editor-post-visibility__dialog-legend">
-							{ __( 'Post Visibility' ) }
-						</div>
-						{ visibilityOptions.map( ( { value, label, info, onSelect, checked } ) => (
-							<label key={ value } className="editor-post-visibility__dialog-label">
+				<span className="editor-post-visibility__button-wrapper">
```

Comment:
Majority of changes here are simply to wrap the `button` and `Popover` in a `span` to which a `position: relative;` style is applied, allowing the `Popover` to position itself within (necessary since `div` is not a valid child of `button`).

Using something like [react-portal](https://github.com/tajo/react-portal) could allow us to be less particular about the DOM structure or relative positioning. Portals are apparently supported in React 16 out-of-the-box, but I am not sure if they are documented yet.